orderedmap/map.go

```diff
@@ -0,0 +1,28 @@
+package orderedmap
+
+import (
+	"constraints"
+	"sort"
+)
+
+// OrderedMap is a map that can be iterated in the natural order of its keys.
+type OrderedMap[K constraints.Ordered, V any] map[K]V
```

Comment:
I don't want this to be long-term, but I can't find any outside container libraries using generics yet. If there's still nothing better in a couple months, we can make our own proper library for this kind of stuff, and I'd much rather this be a bst or rbt to avoid the bad implementation of `Iterate` in that long-term, but this will work for now.

Reply:
This works but +1 on the tree idea. I can see natural progression of testing to have test groups rather than flat ordering.

Reply:
-1 on enforcing an ordering of tests other than the usual `t.Run` stuff happening during an existing test (which we'll have to add support for). running top-level tests in the same order every time is actually not great since you could have unintended dependencies between tests; go recently added the ability to run tests in a random order and I'd like us to start using that whenever possible

additionally, I don't think we even can easily figure out what the "intended order" is registering tests like this, since I don't think there's a guaranteed order to it running the anonymous `init` functions these definitions make

Reply:
for now, it'll run the top-level tests in lexicographical order, and we can support `-shuffle` in the future. if tests need grouping then the normal `t.Run` method can be used after that feature is added (I have it written, will open the review shortly)